transaction/transaction/src/main/java/io/helidon/transaction/Tx.java

```diff
@@ -0,0 +1,303 @@
+import io.helidon.service.registry.Interception;
+import io.helidon.service.registry.Services;
+import io.helidon.transaction.spi.TxSupport;
+
+/**
+ * Transaction annotations and types.
+ */
+public final class Tx {
+
+    private Tx() {
+        throw new UnsupportedOperationException("No instances of Tx are allowed");
+    }
+
+    /**
+     * Transaction type.
+     * Indicates whether method is to be executed within a transaction context where the values provide the following
+     * corresponding behavior.
+     */
+    public enum Type {
+        /**
+         * If called outside a transaction context, the {@link TxException} must be thrown.
+         * If called inside a transaction context, method execution will then continue under that context.
+         */
+        MANDATORY,
+        /**
+         * If called outside a transaction context, the interceptor must begin a new transaction. The managed bean
+         * method execution must then continue inside this transaction context, and the transaction must be completed
+         * by the interceptor.
+         * If called inside a transaction context, the current transaction context must be suspended. New transaction
+         * will begin, the managed method execution must then continue inside this transaction context. The transaction
+         * must be completed, and the previously suspended transaction must be resumed.
+         */
+        NEW,
+        /**
+         * If called outside a transaction context, method execution must then continue outside a transaction context.
+         * If called inside a transaction context, the {@link TxException} must be thrown.
+         */
+        NEVER,
+        /**
+         * If called outside a transaction context, the interceptor must begin a new transaction. The managed bean
+         * method execution must then continue inside this transaction context, and the transaction must be completed
+         * by the interceptor.
+         * If called inside a transaction context, the method execution must then continue inside this transaction context.
+         */
+        REQUIRED,
+        /**
+         * If called outside a transaction context, method execution must then continue outside a transaction context.
+         * If called inside a transaction context, the method execution must then continue inside this transaction context.
+         */
+        SUPPORTED,
+        /**
+         * If called outside a transaction context, method execution must then continue outside a transaction context.
+         * If called inside a transaction context, the current transaction context must be suspended. The method execution
+         * must then continue outside a transaction context, and the previously suspended transaction must be resumed
+         * by the interceptor that suspended it after the method execution has completed.
+         */
+        UNSUPPORTED
+    }
+
+    /**
+     * {@link Type} annotation.
```

Comment:
I'm not sure I understand this.

Reply:
Javadoc comment updated to explain it better.